File: paiement_facilite_app/lib/models/mensualite.dart
class Mensualite {
  final int id;
  final int numero;
  final String dateEcheance;
  final double montant;
  final String statut;
  final String? datePaiement;      // ← ajouter
  final String? photoRecuPath;     // ← ajouter

  Mensualite({
    required this.id,
    required this.numero,
    required this.dateEcheance,
    required this.montant,
    required this.statut,
    this.datePaiement,             // ← ajouter
    this.photoRecuPath,            // ← ajouter
  });

  factory Mensualite.fromJson(Map<String, dynamic> json) {
    return Mensualite(
      id: json['id'],
      numero: json['numero'],
      dateEcheance: json['dateEcheance'],
      montant: (json['montant'] as num).toDouble(),
      statut: json['statut'],
      datePaiement: json['datePaiement'],        // ← ajouter
      photoRecuPath: json['photoRecuPath'],      // ← ajouter
    );
  }
}

File: backend_spring_boot/src/main/java/com/stage/paiement/entity/Mensualite.java
package com.stage.paiement.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.stage.paiement.enums.StatutMensualite;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Mensualite {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer numero;

    private LocalDate dateEcheance;

    private BigDecimal montant;

    @Enumerated(EnumType.STRING)
    private StatutMensualite statut;

    private LocalDate datePaiement;

    private String photoRecuPath;

    @ManyToOne
    @JoinColumn(name = "echeancier_id")
    @JsonBackReference
    private Echeancier echeancier;
}